test/test_pytest_merge_xml.py

```diff
@@ -0,0 +1,156 @@
+            when="call",
+        ),
+        "test_xml_two_failed_one_passed": TestReport(
+            nodeid="test/test_pytest_merge_xml.py::test_xml_two_failed_one_passed",
+            outcome="passed",
+            location=(
+                "test/test_pytest_merge_xml.py",
+                49,
+                "test_xml_two_failed_one_passed",
+            ),
+            keywords="",
+            longrepr=None,
+            when="call",
+        ),
+        "test_xml_three_failed_zero_passed": TestReport(
+            nodeid="test/test_pytest_merge_xml.py::test_xml_three_failed_zero_passed",
+            outcome="failed",
+            location=(
+                "test/test_pytest_merge_xml.py",
+                88,
+                "test_xml_three_failed_zero_passed",
+            ),
+            keywords="",
+            longrepr=None,
+            when="call",
+        ),
+    }
+    obtained_file = datadir / "failure_tmp.xml"
+    modify_xml(Path(datadir / "failure.xml"), retest_results_passed, obtained_file)
+    file_regression.check(obtained_file.read_text(), extension=".xml")
+
+
+def test_xml_two_failed_one_passed(
+    datadir: Path, file_regression: FileRegressionFixture
+) -> None:
+    """Tests modify_xml() with list of flaky test results, with some
+    still not passing, and expects the xml to be overwritten for the
+    passing tests."""
+    retest_results_some_passing = {
+        "test_xml_one_failed_two_passed": TestReport(
+            nodeid="test/test_pytest_merge_xml.py::test_xml_one_failed_two_passed",
+            outcome="failed",
+            location=(
+                "test/test_pytest_merge_xml.py",
+                9,
+                "test_xml_one_failed_two_passed",
+            ),
+            keywords={},
+            longrepr=None,
+            when="call",
+        ),
+        "test_xml_two_failed_one_passed": TestReport(
+            nodeid="test/test_pytest_merge_xml.py::test_xml_two_failed_one_passed",
+            outcome="passed",
+            location=(
+                "test/test_pytest_merge_xml.py",
+                49,
+                "test_xml_two_failed_one_passed",
+            ),
+            keywords="",
+            longrepr=None,
+            when="call",
+        ),
+        "test_xml_three_failed_zero_passed": TestReport(
+            nodeid="test/test_pytest_merge_xml.py::test_xml_three_failed_zero_passed",
+            outcome="failed",
+            location=(
+                "test/test_pytest_merge_xml.py",
+                88,
+                "test_xml_three_failed_zero_passed",
+            ),
+            keywords="",
+            longrepr=None,
+            when="call",
+        ),
+    }
+    obtained_file = datadir / "failure_tmp.xml"
+    modify_xml(
+        Path(datadir / "failure.xml"), retest_results_some_passing, obtained_file
+    )
+    file_regression.check(obtained_file.read_text(), extension=".xml")
+
+
+def test_xml_three_failed_zero_passed(
```

Comment:
Those tests are all valid, but I suggest at least one integration test using the `pytester` fixture.

Reply:
I'll look into this in a later issue! Thanks.